types/express-session/index.d.ts

```diff
@@ -312,7 +319,7 @@ declare namespace session {
     abstract class Store extends EventEmitter {
         regenerate(req: express.Request, callback: (err?: any) => any): void;
         load(sid: string, callback: (err: any, session?: SessionData) => any): void;
-        createSession(req: express.Request, session: SessionData): void;
+        createSession(req: express.Request, session: SessionData): Session & SessionData;
```

Comment:
```suggestion
        createSession(req: express.Request, session: SessionData): Session & Partial<SessionData>;
```

Custom session data has not been initialized here yet.

Reply:
Semi-related: #49479

Reply:
According to [createSession](https://github.com/expressjs/session/blob/1010fadc2f071ddf2add94235d72224cf65159c6/session/store.js#L86) and [session](https://github.com/expressjs/session/blob/1010fadc2f071ddf2add94235d72224cf65159c6/session/store.js#L86), the input session is used to initialize the returned Session object, each attribute is copied into the new object, so I think the type of the input session should be consistent with the return type, they should be both sessionData or both Partial<SessionData>.

Reply:
Oh right of course, I really shouldn't review code in the middle of the night...

In my defense, this seems like another instance of "questionable code" in [`express-session`](https://github.com/expressjs/session/blob/1010fadc2f071ddf2add94235d72224cf65159c6/session/store.js#L86-L102). As far as I can tell there's no reason for that method to be on a `Store`, which makes this whole thing confusing. In the same spirit that it's strange for `cookie` to be on `SessionData`, since `Store`'s don't have anything to do with those, but they are responsible for loading (user-defined) data.

@peterblazejewicz Can you do an extra sanity-check here? This smells a lot like #49479 to me, but I can't quite figure out if these should be `Partial` too 🤔

Reply:
This looks like a merge, partial would make some of props optional, unless I'm missing something.